src/Build/Construction/ProjectElementContainer.cs

```diff
@@ -331,7 +331,7 @@ public void RemoveAllChildren()
         public virtual void DeepCopyFrom(ProjectElementContainer element)
         {
             ErrorUtilities.VerifyThrowArgumentNull(element, nameof(element));
-            ErrorUtilities.VerifyThrowArgument(GetType().IsEquivalentTo(element.GetType()), nameof(element));
+            ErrorUtilities.VerifyThrowArgument(GetType().IsEquivalentTo(element.GetType()), "CannotCopyFromElementOfThatType");
```

Comment:
Could/should this string be passed the type names?

Reply:
if you want? it's probably fairly obscure and to do this I'd have to break out a new VerifyThrowArgumentEquivalentTypes(type, type) because we don't want to create the type name strings unless the condition is false.